renderdoc/driver/d3d12/d3d12_overlay.cpp

```diff
@@ -2089,6 +2089,140 @@ ResourceId D3D12Replay::RenderOverlay(ResourceId texid, FloatVector clearCol, De
   {
     if(pipe && pipe->IsGraphics())
     {
+      ID3D12Resource *renderDepthStencil = renderDepth;
+
+      HRESULT hr;
+      DXGI_FORMAT dsFmt = dsViewDesc.Format;
+      // the depth overlay uses stencil buffer as a mask for the passing pixels
+      if((overlay == DebugOverlay::Depth) && renderDepth)
+      {
+        DXGI_FORMAT dsNewFmt = dsFmt;
+        if(dsFmt == DXGI_FORMAT_D32_FLOAT_S8X24_UINT)
+          dsNewFmt = DXGI_FORMAT_D32_FLOAT_S8X24_UINT;
+        else if(dsFmt == DXGI_FORMAT_D24_UNORM_S8_UINT)
+          dsNewFmt = DXGI_FORMAT_D24_UNORM_S8_UINT;
+        else if(dsFmt == DXGI_FORMAT_D32_FLOAT)
+          dsNewFmt = DXGI_FORMAT_D32_FLOAT_S8X24_UINT;
+        else if(dsFmt == DXGI_FORMAT_D16_UNORM)
+          dsNewFmt = DXGI_FORMAT_D24_UNORM_S8_UINT;
+        else
+          dsNewFmt = DXGI_FORMAT_D32_FLOAT_S8X24_UINT;
+
+        if(m_Overlay.DepthResolvePipe.count(dsNewFmt) == 0)
+        {
+          RDCERR("Unhandled depth resolve format : %s", ToStr(dsNewFmt).c_str());
+          return m_Overlay.resourceId;
+        }
+        if(m_Overlay.DepthCopyPipe.count(dsNewFmt) == 0)
+        {
+          RDCERR("Unhandled depth copy format : %s", ToStr(dsNewFmt).c_str());
+          return m_Overlay.resourceId;
+        }
+        // copy depth over to a new depth-stencil buffer
+        if(dsFmt != dsNewFmt)
+        {
+          D3D12_SHADER_RESOURCE_VIEW_DESC srvDesc = {};
+          srvDesc.Shader4ComponentMapping = D3D12_DEFAULT_SHADER_4_COMPONENT_MAPPING;
+          if(overlayTexDesc.SampleDesc.Count == 1)
+          {
+            srvDesc.ViewDimension = D3D12_SRV_DIMENSION_TEXTURE2D;
+            srvDesc.Texture2D.MipLevels = (UINT)-1;
```

Comment:
This kind of cast looks 'wrong' to me, I'd rather use `~0U` or a #define if there is one.